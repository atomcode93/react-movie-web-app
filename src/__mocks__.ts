export const initialState = {
  auth: {
    error: false,
    isAuthenticated: false,
    user: null,
  },
  favorites: {
    error: false,
    favoritedIds: [],
    favoritedMovies: null,
  },
  filters: {
    order: {
      label: 'Descending',
      value: 'desc',
    },
    rating: {
      max: 10,
      min: 5,
    },
    runtime: {
      max: 250,
      min: 45,
    },
    sort_by: {
      label: 'Rating',
      value: 'vote_average',
    },
    year: 2019,
  },
  modals: {
    args: null,
    open: false,
    type: null,
  },
  movies: {
    error: false,
    favorites: [],
    loading: false,
    movie: null,
    movies: null,
    popularMovies: null,
    soonMovies: null,
    topRatedMovies: null,
    watchLater: null,
  },
  router: {
    action: 'POP',
    location: {
      hash: '',
      pathname: '/',
      search: '',
      state: undefined,
    },
  },
  watchLater: {
    error: false,
    watchLaterIds: [],
    watchLaterMovies: null,
  },
};
