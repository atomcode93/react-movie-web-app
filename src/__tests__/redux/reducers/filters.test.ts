import {
  actions as filtersActions,
  reducer as filtersReducers,
} from 'redux/ducks/filters.duck';
import { initialState } from '__mocks__';

const filtersInitialState = initialState.filters;

const sortBy = {
  value: 'vote_average',
  label: 'Rating',
};

const orderBy = {
  value: 'desc',
  label: 'Descending',
};

const rating = {
  min: 5,
  max: 10,
};
const runtime = {
  min: 45,
  max: 250,
};

describe('Filters Reducer', () => {
  test('should setup default filters state', () => {
    const state = filtersReducers(undefined, { type: '@@INIT' });
    expect(state).toEqual(filtersInitialState);
  });
  test('should update sort filter', () => {
    const action = filtersActions.updateFilters({
      ...initialState.filters,
      sortBy,
    });
    const state = filtersReducers(undefined, action);
    expect(state.sortBy).toMatchSnapshot();
  });
  test('should update order filter', () => {
    const action = filtersActions.updateFilters({
      ...initialState.filters,
      orderBy,
    });
    const state = filtersReducers(undefined, action);
    expect(state.orderBy).toMatchSnapshot();
  });
  test('should update rating filter', () => {
    const action = filtersActions.updateFilters({
      ...initialState.filters,
      rating,
    });
    const state = filtersReducers(undefined, action);
    expect(state.rating).toMatchSnapshot();
  });
  test('should update runtime filter', () => {
    const action = filtersActions.updateFilters({
      ...initialState.filters,
      runtime,
    });
    const state = filtersReducers(undefined, action);
    expect(state.runtime).toMatchSnapshot();
  });
  test('should reset filters', () => {
    const action = filtersActions.resetFilters();
    const state = filtersReducers(undefined, action);
    expect(state).toMatchSnapshot();
  });
});
