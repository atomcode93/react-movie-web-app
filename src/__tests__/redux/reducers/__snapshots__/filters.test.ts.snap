// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters Reducer should reset filters 1`] = `
Object {
  "order": Object {
    "label": "Descending",
    "value": "desc",
  },
  "rating": Object {
    "max": 10,
    "min": 5,
  },
  "runtime": Object {
    "max": 250,
    "min": 45,
  },
  "sort_by": Object {
    "label": "Rating",
    "value": "vote_average",
  },
  "year": 2019,
}
`;

exports[`Filters Reducer should update order filter 1`] = `
Object {
  "label": "Descending",
  "value": "desc",
}
`;

exports[`Filters Reducer should update rating filter 1`] = `
Object {
  "max": 10,
  "min": 5,
}
`;

exports[`Filters Reducer should update runtime filter 1`] = `
Object {
  "max": 250,
  "min": 45,
}
`;

exports[`Filters Reducer should update sort filter 1`] = `
Object {
  "label": "Rating",
  "value": "vote_average",
}
`;
